src/main/java/com/streamr/client/utils/BigDecimalAdapter.java

```diff
@@ -0,0 +1,25 @@
+package com.streamr.client.utils;
+
+import com.squareup.moshi.JsonAdapter;
+import com.squareup.moshi.JsonReader;
+import com.squareup.moshi.JsonWriter;
+
+import java.io.IOException;
+import java.math.BigDecimal;
+
+public class BigDecimalAdapter extends JsonAdapter<BigDecimal> {
+    @Override
+    public BigDecimal fromJson(JsonReader reader) throws IOException {
+        if (reader.peek() == JsonReader.Token.NULL) return reader.nextNull();
```

Comment:
Can probably remove the null-checking here and in `toJson` if you use `new BigDecimalAdapter().nullSafe()` instead of just `new BigDecimalAdapter()`.